### PR TITLE
fix that snaps objects onto the table surface in rnt-painting

### PR DESCRIPTION
**Pull request by NishanthJKumar**
necessary to get learned samplers to work in this environment!

### CODE REVIEW

File: src/envs/painting.py [hunk 1]
```diff
@@ -285,7 +285,17 @@ def _transition_place(self, state: State, action: Action) -> State:
         next_state.set(held_obj, "pose_x", x)
         next_state.set(held_obj, "pose_y", y)
         if self._update_z_poses:
-            next_state.set(held_obj, "pose_z", z)
+            if receptacle == "table" and np.allclose(
+                    z, self.table_height + self.obj_height / 2, rtol=5e-02):
```
Comment (tomsilver) [inline src/envs/painting.py:289]: can we make this constant a class variable?

**Review by tomsilver — COMMENTED** [review]
thanks for finding and fixing this! could you also add a unit test that fails before this change and succeeds after it?

**Review by tomsilver — APPROVED** [review]
lgtm after the the unit tests pass!

**Review by ronuchit — APPROVED** [review]
great catch!

**Pull request merged**